app/services/process_state.rb

```diff
@@ -0,0 +1,45 @@
+# ProcessState is an *experimental* thing that infers the state from
```

Comment:
It's not easy to understand what this service does from its name `ProcessState`.
What about 
`GetApplicationFormState`?